### PR TITLE
Plugin features

### PR DESCRIPTION
**Pull request by Erunx**

### CODE REVIEW

File: lang/en/enrol_payment.php [hunk 1]
```diff
@@ -3,3 +3,4 @@
 $string['pluginname'] = 'Payment software';
 $string['pluginname_desc'] = 'Empty skeleton of the enrolment playment plugin.';
```
Comment (harunaka11037) [inline lang/en/enrol_payment.php]: Typo + outdated

File: lib.php [hunk 2]
```diff
@@ -4,30 +4,19 @@
...
-            //returns "This course is free."
-            echo '<p>'.get_string('nocost', 'enrol_payment').'</p>';
-        }
-
-        
         
-            
-        //stops saving 
-        //ob_end_clean();
+        echo "<p><b>To enrol into a class please pay the amount below</b></p>";
```
Comment (harunaka11037) [inline lib.php]: Might be better to use get_string for language support

File: version.php [hunk 3]
```diff
@@ -2,11 +2,11 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$plugin->version   = 2019111802;        // The current plugin version (Date: YYYYMMDDXX)
+$plugin->version   = 2019111807;        // The current plugin version (Date: YYYYMMDDXX)
```
Comment (harunaka11037) [inline version.php]: Might be a good idea to update version number to a new date (I believe the last XX bit in the YYYYMMDDXX format is just 00 for the first release that day, and increases by 1 for all subsequent releases during that day)

**Review by harunaka11037 — CHANGES_REQUESTED** [review]

File: db/access.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
+<?php
```
Comment (Peterburnett) [inline db/access.php:1]: Moodle Boilerplate header here

File: db/access.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    /* Unenrol self - watch out for data loss. */
+    'enrol/payment:unenrolself' => array(
+        'captype' => 'write',
+        'contextlevel' => CONTEXT_COURSE,
+        'archetypes' => array(
+        )
+    ),
+
+);
```
Comment (Peterburnett) [inline db/access.php:76]: Add a newline here. Should be able to config this in your IDE so it autoadds newlines at the end of files

File: lib.php [hunk 6]
```diff
@@ -4,30 +4,25 @@
 defined('MOODLE_INTERNAL') || die();
 class enrol_payment_plugin extends enrol_plugin {
 
-    public function enrol_page_hook(stdClass $instance){  
-        global $CFG, $USER, $OUTPUT, $PAGE, $DB;
+     public function enrol_page_hook(stdClass $instance){  
+         global $CFG, $USER, $OUTPUT, $PAGE, $DB;
 
-        //saves everything about to happen
         ob_start();
```
Comment (Peterburnett) [inline lib.php]: Dont use output buffering here. Instead write all HTML into a string, which you keep appending to, then echo the string at the end

File: lib.php (hunk 2 above)
Comment (Peterburnett) [reply lib.php]: Also all of these HTML tags should be escaped using \html_writer::tag('p', $string);

File: lib.php [hunk 7]
```diff
@@ -4,30 +4,25 @@
...
+        echo "<p><b>To enrol into a class please pay the amount below</b></p>";
 
-        //Check the price of the course 
-        if (abs($cost) < 0.01) { 
-            //returns "This course is free."
-            echo '<p>'.get_string('nocost', 'enrol_payment').'</p>';
-        }
+        echo "<p>Price: $</p>";
 
+        if(array_key_exists('paymentButton',$_POST)){
```
Comment (Peterburnett) [inline lib.php]: Dont access POST directly, big security issue. Use optional_param or required_param. They can check POST as well. They validate it for safety, and strip out nasties

File: lib.php [hunk 8]
```diff
@@ -4,30 +4,25 @@
...
+            echo "<p>button selected</p>";
+        }
+        
+        echo  '<button onClick="window.location=\'' . $CFG->wwwroot . '\';">Enrol Now!</button>';
```
Comment (Peterburnett) [inline lib.php]: Make this into a \single_button instead. Does this work for you

File: lib.php [hunk 9]
```diff
@@ -36,30 +31,41 @@ public function enrol_page_hook(stdClass $instance){
     
 
 
-    public function roles_protected() {
-        // Return true if plugin allows manual modification of user enrolments from other code. 
-        //False is usually returned from plugins that synchronise data with external systems, 
-        //otherwise the manual changes would be reverted immediately upon synchronisation.
-        return false;
+    // public function roles_protected($instance) {
+    //     // Return true if plugin allows manual modification of user enrolments from other code.
```
Comment (Peterburnett) [inline lib.php]: Make this into a PHPDoc block, otherwise Codechecker will complain

Comment (Peterburnett) [reply lib.php]: Actually just delete the commented stuff. No point keeping it

File: lib.php [hunk 10]
```diff
@@ -102,6 +107,6 @@ public function edit_instance_validation($data, $files, $instance, $context) {
      * @return boolean                                                                                                              
      */                                                                                                                             
     public function can_add_instance($courseid) {  
-
+        return true;
     }
 }
```
Comment (Peterburnett) [inline lib.php:110]: Another newline issue here

File: setting.php [hunk 11]
```diff
@@ -1,4 +1,8 @@
```
Comment (Peterburnett) [inline setting.php]: Just kill this file if not using it

File: lang/en/enrol_payment.php [hunk 12]
```diff
@@ -1,5 +1,6 @@
 <?php
```
Comment (Peterburnett) [inline lang/en/enrol_payment.php:1]: Moodle boilerplate header here

File: lib.php [hunk 13]
```diff
@@ -4,30 +4,25 @@
 defined('MOODLE_INTERNAL') || die();
```
Comment (Peterburnett) [inline lib.php:26]: Moodle boilerplate header here

File: version.php [hunk 14]
```diff
@@ -2,11 +2,11 @@
 
 defined('MOODLE_INTERNAL') || die();
```
Comment (Peterburnett) [inline version.php:26]: Moodle boilerplate heading here

**Review by Peterburnett — CHANGES_REQUESTED** [review]

**Pull request merged**